### PR TITLE
GOONCHAT

### PR DESCRIPTION
**Pull request by ghost**
:cl: ErikHanson
experiment: Ported goon's browser based chat, this has a ton of advantages. <b>CREDITS TO WIRE (goon coder) FOR A TON OF THE WORK</b>
experiment: This might break some things, so please report them on our github!
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
what does it actually do

**Comment by ghost** [comment]
@cacogen it ports this: https://github.com/d3athrow/vgstation13/pull/9043 which is basically a port of Goonstations chat system written by Wire

**Comment by cacogen** [comment]
there was a lot of fallout in that pr after it was merged

are the problems they mention there fixed here?

**Comment by ghost** [comment]
Correct

**Comment by cacogen** [comment]
it's still not clear what the advantages of this are other than smoother font rendering

**Comment by tortellinitony** [comment]
please no

**Comment by JohnGinnane** [comment]
>1,054 files changes

okay

File: code/__HELPERS/_logging.dm [hunk 1]
```diff
@@ -90,8 +90,7 @@
 //This replaces world.log so it displays both in DD and the file
 /proc/log_world(text)
```
[the remark this answers is not in the document]
Comment (ghost) [reply code/__HELPERS/_logging.dm]: > //This replaces world.log so it displays both in DD and the file

might have been a leftover from another branch of mine, i'll take a look once im done with work today

**Comment by ghost** [comment]
@tortellinitony why not? :-)

**Comment by j-mie** [comment]
I'd rather port this from tg's port tbh, that way we're running as identical code as possible

**Comment by ghost** [comment]
@JamieH [this is the /tg/ port /tg/ basically ported the /vg/ one and made some rather small changes](https://patch-diff.githubusercontent.com/raw/tgstation/tgstation/pull/24127.patch)

I just pulled [this](https://patch-diff.githubusercontent.com/raw/tgstation/tgstation/pull/24127.patch) patchfile, and merged it locally and fixed the rest by hand

**Comment by j-mie** [comment]
Admin PM doesn't even work...

**Comment by ghost** [comment]
I'll pick this back up once /tg/ has merged theirs

**Pull request closed without merging**